src/TestItemRunner2.jl

```diff
@@ -132,12 +132,20 @@ function get_free_testprocess(testitem, max_num_processes)
     end
 end
 
-function execute_test(test_process, testitem, testsetups)
+function execute_test(test_process, testitem, testsetups, timeout)
```

Comment:
@nickrobinson251 / @quinnj / @drvi and I had been talking about possibly introducing a timeout parameter on the testitems themselves.

What do you think about on the testitem vs on the run_tests call? Maybe we'd want both?

Reply:
Yeah, I think both probably makes sense... The only thing I don't like about that is that timeouts generally seem somewhat machine specific (at least in some cases), and that doesn't seem to fit having them on the test items. But on the other hand, there are probably timeouts that are not of that nature where attaching them to a specific test item does make sense? Say a test that downloads something?